src/Control/Retry.hs

```diff
@@ -96,27 +105,38 @@ import           Prelude                hiding (catch)
 -- For anything more complex, just define your own 'RetryPolicy':
 --
 -- >> myPolicy = RetryPolicy $ \ n -> if n > 10 then Just 1000 else Just 10000
-newtype RetryPolicy = RetryPolicy { getRetryPolicy :: Int -> Maybe Int }
+newtype RetryPolicyM m = RetryPolicyM { getRetryPolicyM :: Int -> m (Maybe Int) }
 
 
-instance Default RetryPolicy where
+-- | 'RetryPolicyM' without monadic logic in determining the reply.
+type RetryPolicy = RetryPolicyM Identity
```

Comment:
I'd rather define `type RetryPolicy = forall m. Monad m => RetryPolicyM m`. This way we can leave most of the type signatures as they are and use `RetryPolicyM` only in `fullJitterBackoff`.